sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -439,6 +439,14 @@ object StateStore extends Logging {
   @GuardedBy("loadedProviders")
   private val schemaValidated = new mutable.HashMap[StateStoreProviderId, Option[Throwable]]()
 
+  private val threadPoolLock = new Object
+
+  @GuardedBy("threadPoolLock")
+  private var threadPoolException: Throwable = null
+
+  @GuardedBy("threadPoolLock")
+  private val partitions = new mutable.HashSet[StateStoreProviderId]
```

Comment:
nit: the name is too general to indicate what it will be used. Could we please either add code comment to explain or rename it to be slightly verbose?